google/cloud/internal/backoff_policy.h

```diff
@@ -126,12 +126,19 @@ class ExponentialBackoffPolicy : public BackoffPolicy {
   ExponentialBackoffPolicy(std::chrono::duration<Rep1, Period1> initial_delay,
                            std::chrono::duration<Rep2, Period2> maximum_delay,
                            double scaling)
-      : initial_delay_(std::chrono::duration_cast<std::chrono::microseconds>(
-            initial_delay)),
-        current_delay_range_(2 * initial_delay_),
-        maximum_delay_(std::chrono::duration_cast<std::chrono::microseconds>(
-            maximum_delay)),
-        scaling_(scaling) {
+      : initial_delay_(
+            std::chrono::duration_cast<
+                std::chrono::duration<double, std::micro>>(initial_delay)),
+        maximum_delay_(
+            std::chrono::duration_cast<
+                std::chrono::duration<double, std::micro>>(maximum_delay)),
+        scaling_(scaling),
+        current_delay_start_(initial_delay_),
+        current_delay_end_((std::min)(
+            std::chrono::duration_cast<
+                std::chrono::duration<double, std::micro>>(scaling_ *
+                                                           initial_delay_),
+            maximum_delay_)) {
```

Comment:
nit: now that we do the bounds checking before generating the random number, we don't need to do a min here. We can just say it = `initial_delay_ * scaling_`